cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -350,7 +350,7 @@ def mapped_op(self, deep: bool = False) -> 'cirq.CircuitOperation':
     def _decompose_(self) -> Iterator['cirq.Operation']:
         return self.mapped_circuit(deep=False).all_operations()
 
-    def _act_on_(self, args: 'cirq.OperationTarget') -> bool:
+    def _act_on_(self, args: 'cirq.SimulationStateBase') -> bool:
```

Comment:
I did for local variables. For function arguments I expect there'll need to be a deprecation cycle. Though I guess for private / protocol methods that doesn't apply.

Reply:
I went ahead and updated it for the protocol handlers. I think I'm going to have to take a rain check on the rest as there's just too much to fit in one PR, and `args` is way too common to find and replace. The main one is `_create_partial_act_on_args`, which is a bit complicated and will require a deprecation cycle and probably some dynamic hasattr or inspect stuff to be backwards compatible. Beyond that, misc stuff here and there, mostly in `sim`.. Was there anything else in particular you wanted me to deal with in this PR?

Reply:
Good call on the deprecation cycle. We should keep an eye on it even for the protocol methods, since they have a bad habit of being called from outside of Cirq.

I don't see anything else urgent to deal with in this PR - moving to #5296 for further review.